File: C++/1040.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main()
{
    double N1, N2, N3, N4, media, exame, notafinal;

    cin >> N1 >> N2 >> N3 >> N4;
media= ((N1*2)/10) + ((N2*3)/10) + ((N3*4)/10)+ ((N4*1)/10);
    cout << fixed << setprecision(1) << "Media: " << media<< endl;
    if (media>= 7)
    {
        cout << "Aluno aprovado." << endl;
    }
    else if (media<5)
    {
        cout << "Aluno reprovado." << endl;
    }
    else
    {
        cout << "Aluno em exame." << endl;

        cin >> exame;
        cout << fixed << setprecision(1) << "Nota do exame: " << exame << endl;
        notafinal= (exame+media)/2;

        if (notafinal>=5 ) {
            cout << "Aluno aprovado." << endl;
        }
else{ cout << "Aluno reprovado." << endl;
    }


 
     cout <<fixed << setprecision(1) << "Media final: " << notafinal << endl;
    }

         return 0;
}

File: Python/1040.py
notas = input().split()

n1 = float(notas[0])
n2 = float(notas[1])
n3 = float(notas[2])
n4 = float(notas[3])

media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / 10

print(f'Media: {media:.1f}')

if media >= 5.0 and media <= 6.9:
    print('Aluno em exame.')
    recuperacao = float(input())
    medRep = (recuperacao + media) / 2
    
    print(f'Nota do exame: {recuperacao:.1f}')

    if medRep >= 5.0:
        print('Aluno aprovado.')
        print(f'Media final: {medRep:.1f}')
    else:
        print('Aluno reprovado.')

elif media >= 7.0:
    print('Aluno aprovado.')

else:
    print('Aluno reprovado.')
